autogpt_platform/backend/backend/blocks/slack/_oauth.py

```
@@ -0,0 +1,163 @@
+    """
+    OAuth2 handler for Slack's bot-install flow.
+
+    Classic Slack bot tokens (``xoxb-...``) never expire and have no refresh
+    token. Only workspaces where the installing app has opted into "token
+    rotation" issue expiring tokens with a ``refresh_token`` — this handler
+    supports both by treating refresh as a no-op when there's nothing to
+    refresh. Slack's OAuth v2 endpoint does not support PKCE.
+    """
+
+    PROVIDER_NAME = ProviderName.SLACK
+    DEFAULT_SCOPES = ["chat:write"]
```

Ruff (0.16.0)

[warning] Mutable default value for class attribute

(RUF012)